.github/workflows/cli_commands.yml

```diff
@@ -15,236 +15,344 @@ jobs:
+                          ACTUAL_WASI2IC_VERSION=$(echo "${WASI2IC_LINE}" | sed -E 's/.*wasi2ic[[:space:]]+v([0-9]+\.[0-9]+\.[0-9]+).*/\1/')
+                        fi
+                      else
+                        ACTUAL_WASI2IC_VERSION="not installed"
+                      fi
+                    else
+                      ACTUAL_WASI2IC_VERSION="not installed"
+                    fi
+
+                    echo "Actual wasi2ic version: ${ACTUAL_WASI2IC_VERSION}"
+                    if [[ "${ACTUAL_WASI2IC_VERSION}" == "${EXPECTED_WASI2IC_VERSION}" ]]; then
+                      echo "✅ wasi2ic version verification successful"
+                    else
+                      echo "❌ wasi2ic version verification failed: Expected ${EXPECTED_WASI2IC_VERSION}, got ${ACTUAL_WASI2IC_VERSION}"
+                      exit 1
+                    fi
+                  fi
+
+                  echo "✅ All version verifications completed successfully"
+
             - name: Test azle dev template commands
-              working-directory: azle
               run: |
                   echo "Testing azle dev template commands"
 
+                  # Helper function to clean all template files
+                  clean_templates() {
+                    echo "Cleaning all template files"
+                    rm -f dist/canister_templates/stable.wasm
+                    rm -f dist/canister_templates/experimental.wasm
+                  }
+
+                  # Helper function to check file size
+                  check_file_size() {
+                    local file="$1"
+                    local file_size=$(stat -c%s "$file" 2>/dev/null || echo "0")
+                    if [[ "$file_size" -gt 0 ]]; then
+                      return 0
+                    else
+                      return 1
+                    fi
+                  }
+
                   # Remove existing template files to test regeneration
-                  echo "Removing existing template files for regeneration test"
-                  rm -f dist/canister_templates/stable.wasm
-                  rm -f dist/canister_templates/experimental.wasm
+                  echo "Initial cleanup: Removing existing template files for regeneration test"
+                  clean_templates
 
                   # Test azle dev template (stable)
-                  echo "Testing 'azle dev template' command"
+                  echo "Testing 'azle dev template' command (stable only)"
                   npx azle dev template
 
-                  # Verify stable template was created
+                  # Verify stable template was created and has valid size
                   if [[ -f "dist/canister_templates/stable.wasm" ]]; then
-                    echo "✅ Stable template verification successful: stable.wasm was created"
+                    echo "✅ Stable template file created successfully"
+                    if ! check_file_size "dist/canister_templates/stable.wasm"; then
```

Comment:
Make sure the `!` and the return values of `0` and `1` work appropriately

Reply:
Yes, when you return from a bash function it's the same as an exit code. The function is successful if the file has size, it's unsuccessful if it has no size. So if the check_file_size check is unsuccessful then we fail this test.